core/genesis.go

```diff
@@ -350,13 +356,17 @@ func (g *Genesis) Commit(db ethdb.Database) (*types.Block, error) {
 	if err := config.CheckConfigForkOrder(); err != nil {
 		return nil, err
 	}
-	rawdb.WriteBlock(db, block)
-	rawdb.WriteReceipts(db, block.Hash(), block.NumberU64(), nil)
-	rawdb.WriteCanonicalHash(db, block.Hash(), block.NumberU64())
-	rawdb.WriteHeadBlockHash(db, block.Hash())
-	rawdb.WriteHeadHeaderHash(db, block.Hash())
-	rawdb.WriteChainConfig(db, block.Hash(), config)
-	rawdb.WriteUpgradeConfig(db, block.Hash(), &config.UpgradeConfig)
+	batch := db.NewBatch()
```

Comment:
ooh that probably also goes to coreth + even geth?

Reply:
Ya I don't think we've ever a report of this happening, but there was a warn log in there to explicitly handle one case where the genesis block is written, but the chain config hasn't been written yet, which can be avoided by writing the block and chain config together on one batch.